diff --git a/cmd-pipe-pane.c b/cmd-pipe-pane.c
index 0fa656ce..ddd5e998 100644
--- a/cmd-pipe-pane.c
+++ b/cmd-pipe-pane.c
@@ -109,10 +109,17 @@ cmd_pipe_pane_exec(struct cmd *self, struct cmdq_item *item)
 	}
 
 	/* Open the new pipe. */
+#ifndef __MVS__
 	if (socketpair(AF_UNIX, SOCK_STREAM, PF_UNSPEC, pipe_fd) != 0) {
 		cmdq_error(item, "socketpair error: %s", strerror(errno));
 		return (CMD_RETURN_ERROR);
 	}
+#else
+	if (pipe(pipe_fd) != 0) {
+		cmdq_error(item, "pipe error: %s", strerror(errno));
+		return (CMD_RETURN_ERROR);
+	}
+#endif
 
 	/* Expand the command. */
 	ft = format_create(cmdq_get_client(item), item, FORMAT_NONE, 0);
