diff --git a/environ.c b/environ.c
index 74d672e0..e4971fe0 100644
--- a/environ.c
+++ b/environ.c
@@ -15,13 +15,14 @@
  * IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
  * OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-
+#define environ environ_replace
 #include <sys/types.h>
 
 #include <fnmatch.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#undef environ
 
 #include "tmux.h"
 
